packages/data-provider/src/repository/submittedRepository.ts

```diff
@@ -130,11 +144,15 @@ const repository = (dependencies: BaseDependencies) => {
 		getSubmittedDataByCategoryIdPaginated: async (
 			categoryId: number,
 			paginationOptions: PaginationOptions,
+			filter?: { entityNames?: string[] },
 		): Promise<SubmittedDataResponse[]> => {
 			const { page, pageSize } = paginationOptions;
+
+			const filterEntityNameSql = filterByEntityNameArray(filter?.entityNames);
+
 			try {
 				return await db.query.submittedData.findMany({
-					where: eq(submittedData.dictionaryCategoryId, categoryId),
+					where: and(eq(submittedData.dictionaryCategoryId, categoryId), filterEntityNameSql),
```

Comment:
how does this work when `filterEntityNameSql` is `undefined`? any issues? this will happen when there are no entityName query params provided.

Reply:
Based on some other uses of the drizzle `and()` it looks like `undefined` is commonly used like this, so no worries here :+1:

Reply:
there is no issues when filterEntityNameSql is `undefined` Drizzle handles it with no issues.